src/main/scala/io/iohk/ethereum/network/p2p/messages/PV63.scala

```diff
@@ -190,7 +195,12 @@ object PV63 {
     implicit class ReceiptRLPEncodableDec(val rlpEncodeable: RLPEncodeable) extends AnyVal {
       def toReceipt: Receipt = rlpEncodeable match {
         case RLPList(postTransactionStateHash, cumulativeGasUsed, logsBloomFilter, logs: RLPList) =>
-          Receipt(postTransactionStateHash, cumulativeGasUsed, logsBloomFilter, logs.items.map(_.toTxLogEntry))
+          val stateHash = postTransactionStateHash match {
+            case RLPValue(bytes) if bytes.length > 1 => HashOutcome(ByteString(bytes))
+            case RLPValue(bytes) if bytes.length == 1 && bytes.head == 1 => SuccessOutcome
+            case _ => FailureOutcome
```

Comment:
I prefer to write explicit all meaningful cases. In this situation value different than RLPValue is treated as a FailureOutcome.